packages/api/amplify_api/lib/src/graphql/graphql_send_request.dart

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../util.dart';
+import 'graphql_response_decoder.dart';
+
+@internal
+Future<GraphQLResponse<T>> GraphQLSendRequest<T>(
+    {required GraphQLRequest<T> request,
+    required http.Client client,
+    required Uri uri}) async {
+  final body = {'variables': request.variables, 'query': request.document};
+  final graphQLResponse = await client.post(uri, body: json.encode(body));
+
```

Comment:
I think there is room for improvement here regarding error handling/robustness. What if response is not a 200? What if JSON does not have expected structure? Would be good to have a look at other platforms and get an idea how this should work. Also suggest adding tests for that logic.

Reply:
The [spec](https://spec.graphql.org/June2018/#sec-Response-Format) implies that a 200 response guarantees a certain shape to the data. We can still verify that we get a Map back, but we must check if the status code is 200 first. Any other code means something went wrong.